src/app/Components/Details/RentTransactions/index.jsx

```diff
@@ -1,21 +1,32 @@
 import React, { Component } from 'react';
+import './index.scss';
 
 export default class RentTransactions extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      transactionsDisplayed: 10
+      transactionsDisplayed: 20,
+      pageNumber: 0
     };
   }
 
   render() {
-    const transactions = this.props.transactions
-      .slice(0, this.state.transactionsDisplayed)
-      .map(transaction => {
-        return transaction;
-      });
+    let resultsRange = [];
 
-    if (transactions.length > 0) {
+    const totalTransactions = this.props.transactions.length;
+    const firstPageItem =
+      this.state.pageNumber * parseInt(this.state.transactionsDisplayed);
+    const remainingTransactions = totalTransactions - firstPageItem;
+    const lastPageItem =
+      remainingTransactions < parseInt(this.state.transactionsDisplayed)
+        ? remainingTransactions + firstPageItem
+        : firstPageItem + parseInt(this.state.transactionsDisplayed);
+
+    for (let i = firstPageItem; i < lastPageItem; i++) {
```

Comment:
I think this could be a lot simpler using `array.slice` which will mean you don't need to check how many elements to loop through. e.g.

```
resultsRange = this.props.transactions.slice(this.state.pageNumber * this.state.transactionsDisplayed, this.state.transactionsDisplayed)
```